data-prepper-plugins/saas-source-plugins/crowdstrike-source/src/main/java/org/opensearch/dataprepper/plugins/source/crowdstrike/CrowdStrikeService.java

```diff
@@ -44,35 +47,30 @@ public CrowdStrikeService(CrowdStrikeRestClient crowdStrikeRestClient, PluginMet
      *   @param paginationLink  An optional pagination URL suffix (used when fetching next pages).
      *   @return                A {@link CrowdStrikeApiResponse} containing response body and headers.
      */
-    public CrowdStrikeApiResponse getAllContent(Long startTime, Long endTime, String paginationLink) {
+    public CrowdStrikeApiResponse getThreatIndicators(Instant startTime, Instant endTime, Optional<String> paginationLink) {
+        if (startTime == null || endTime == null) {
+            throw new IllegalArgumentException("startTime and endTime must not be null");
+        }
         URI uri = buildCrowdStrikeUri(startTime, endTime, paginationLink);
-
         return searchCallLatencyTimer.record(() -> {
-            try {
-                log.debug("Calling CrowdStrike API with URI: {}", uri);
-                ResponseEntity<CrowdStrikeIndicatorResult> responseEntity = crowdStrikeRestClient.invokeGetApi(uri, CrowdStrikeIndicatorResult.class);
 
-                CrowdStrikeApiResponse response = new CrowdStrikeApiResponse();
-                response.setBody(responseEntity.getBody());
-                response.setHeaders(responseEntity.getHeaders());
-                return response;
-            } catch (Exception e) {
-                log.error("Error fetching CrowdStrike content from URI: {}", uri, e);
-                throw new RuntimeException("CrowdStrike API call failed", e);
-            }
+            log.debug("Calling CrowdStrike API with URI: {}", uri);
+            ResponseEntity<CrowdStrikeIndicatorResult> responseEntity = crowdStrikeRestClient.invokeGetApi(uri, CrowdStrikeIndicatorResult.class);
+
+            return new CrowdStrikeApiResponse(responseEntity.getBody(), responseEntity.getHeaders());
         });
     }
 
-    protected URI buildCrowdStrikeUri(Long startTime, Long endTime, String paginationLink) {
+    protected URI buildCrowdStrikeUri(Instant startTime, Instant endTime,  Optional<String> paginationLink) {
         try {
-            if (paginationLink != null) {
-                String urlString = BASE_URL + paginationLink;
+            if (paginationLink.isPresent()) {
+                String urlString = BASE_URL + paginationLink.get();
                 urlString = CrowdStrikeNextLinkValidator.validateAndSanitizeURL(urlString);
                 return new URI(urlString);
             } else {
                 // Manually construct and encode the query string
-                String filter1 = URLEncoder.encode(LAST_UPDATED + ":>=" + startTime, StandardCharsets.UTF_8);
-                String filter2 = URLEncoder.encode(LAST_UPDATED + ":<" + endTime, StandardCharsets.UTF_8);
+                String filter1 = URLEncoder.encode(LAST_UPDATED + ":>=" + startTime.getEpochSecond(), StandardCharsets.UTF_8);
+                String filter2 = URLEncoder.encode(LAST_UPDATED + ":<" + endTime.getEpochSecond(), StandardCharsets.UTF_8);
```

Comment:
minor: startTimeFilter, endTimeFilter would be better names than filter1, filter2